sympy/core/basic.py

```diff
@@ -1557,12 +1557,24 @@ def doit(self, **hints):
             return self
 
     def _eval_rewrite(self, pattern, rule, **hints):
+        from sympy.simplify.simplify import simplify
         if self.is_Atom:
             if hasattr(self, rule):
                 return getattr(self, rule)()
             return self
 
-        if hints.get('deep', True):
+        if rule == '_eval_rewrite_as_sinc':
```

Comment:
This logic should be handled outside of the rewrite framework. If anything, if you want nested rewrites to happen automatically that should be done in a more generic way. I don't like one-off hacks like this, especially in the core. 

You can just change the test you added to `F.rewrite(sin).rewrite(sinc)`, or add `exp._eval_rewrite_sinc`.